packages/vscode/src/utils/git.util.ts

```diff
@@ -194,4 +194,37 @@ export const getRepo = (gitRemoteConfig: string) => {
 	}
 
 	return { owner, repo };
-}
\ No newline at end of file
+}
+
+export async function getBranchNames(path: string, repo: string): Promise<string[]> {
+	return new Promise((resolve, reject) => {
+		resolveSpawnOutput(
+			cp.spawn(path, ["branch"], {
+				cwd: repo,
+				env: Object.assign({}, process.env),
+			})
+		).then((values) => {
+			const status = values[0], stdout = values[1], stderr = values[2];
+			if (status.code === 0) {
+				const branches = stdout
+					.toString()
+					.split("\n")
+					.map((name) => name.replace("*", "").trim());
+				resolve(branches);
+			} else {
+				reject(stderr);
+			}
+		});
+	});
+}
+
+export function getBaseBranchName(branchNames: string[]): string {
+	for (const name of branchNames) {
+		if (name === "main") {
+			return "main";
+		} else if (name === "master") {
+			return "master";
+		}
+	}
+	return branchNames[0];
```

Comment:
(물론 잘 알고계시겠지만..)
개인적으로는 목록에서 뭔가를 가져올때는 Array.prototype.find 로 표현하는게 좋아보이는데.. 요거 어떻게 생각하시나요ㅎㅎ (이 케이스같은 경우는 루프순회 횟수도 동일하겠네요)

```suggestion
	const baseBranch = branchNames.find((n) => n === "main" | n === "master")
	                // branchNames.find((n) => ['main','master'].includes(n))
	    ?? branchNames[0];
	return baseBranch
```

Reply:
find도 깔끔하게 좋네요!!